### ISSUE TITLE
Allow to delete keys by the prefix

### ISSUE THREAD

**Issue by Taymindis** [apache/kvrocks#2628]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

- Using Namespace in cluster mode to easily flush the keys .

### Solution

- As i knew namespace simply prepend the namespace prefix to the user key and remove it before retrieving. But it couldn't achieve with just simple key prefix.

we use the hash bracket like below, if our namespace is `foo`, and key is 1234, it will be successfully removed all the keys with same namespace.
{foo}:1234

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by git-hulk** [comment]
@Taymindis Delete keys with the specified prefix would be easy in Kvrocks, but Redis now doesn't have a command for this and we need to invent one if we would like to support this. What do you think? @apache/kvrocks-committers

**Comment by ShooterIT** [comment]
As all keys are sorted by lexicographic order in kvrocks, maybe we can think of a set of commands about key prefix operation, of course, deleting keys by the prefix.
oh `scan` already supports key pattern, but kvrocks is faster than expected

**Comment by git-hulk** [comment]
> oh scan already supports key pattern, but kvrocks is faster than expected

@ShooterIT Yes, another fast path is to use `DeleteRange` for the deletion command.

**Comment by baobaomaomeng** [comment]
Does it need to have a limit on the number of matches, similar to prefix matching?And it‘s just imple for keyType？

**Comment by PragmaTwice** [comment]
Several questions:
- It seems there's two PRs for this one task, so it's quite confusing. Are these two PRs competing with each other?
- I think first we should define the semantics and command syntax of it. It will affect the performance of this command.
